AsyncDisplayKit/Details/ASRangeControllerBeta.mm

```diff
@@ -225,6 +266,33 @@ - (void)_updateVisibleNodeIndexPaths
 #endif
 }
 
+#pragma mark - Notification observers
+
+- (void)registerForNotificationsIfNeeded
+{
+  if (!_didRegisterForNotifications) {
+    BOOL selfInterfaceState = [_dataSource interfaceStateForRangeController:self];
+    ASLayoutRangeMode nextRangeMode = [ASRangeControllerBeta rangeModeForInterfaceState:selfInterfaceState
+                                                                        scrollDirection:_scrollDirection
+                                                                       currentRangeMode:_currentRangeMode];
+    if (_currentRangeMode != nextRangeMode) {
+      [[NSNotificationCenter defaultCenter] addObserver:self
+                                               selector:@selector(scheduledNodesDidDisplay)
+                                                   name:ASRenderingEngineDidDisplayScheduledNodesNotification
+                                                 object:nil];
+      _didRegisterForNotifications = YES;
+    }
+  }
+}
+
+- (void)scheduledNodesDidDisplay
+{
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
```

Comment:
Since this is unreleased and just in master, I am not worried at the moment, but have you audited this for any possible race conditions? Is it possible that another range controller could be operating at the same time, Such as two scroll views next to one another on iPad, or that a prior display operation could trigger this method to be called and cause us to and observe right after we complete an editing transaction that kicks off more node loading?